tee-worker/litentry/core/parachain-extrinsic-task/receiver/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use itp_extrinsics_factory::CreateExtrinsics;
+use itp_ocall_api::EnclaveOnChainOCallApi;
+use itp_types::parentchain::ParentchainId;
+use lc_parachain_extrinsic_task_sender::init_parachain_extrinsic_sender_storage;
+use log::*;
+use std::{format, string::String, sync::Arc, time, vec};
+
+const MAX_BATCH_SIZE: usize = 500;
+const BATCH_EXTRINSIC_INTERVAL: time::Duration = time::Duration::from_secs(6);
+
+pub fn run_parachain_extrinsic_task_receiver<ExtrinsicsFactory, OCallApi>(
+	api: Arc<OCallApi>,
+	extrinsic_factory: Arc<ExtrinsicsFactory>,
+) -> Result<(), String>
+where
+	ExtrinsicsFactory: CreateExtrinsics + Send + Sync + 'static,
+	OCallApi: EnclaveOnChainOCallApi,
+{
+	let task_receiver = init_parachain_extrinsic_sender_storage().map_err(|e| {
+		format!("Failed to initialize parachain extrinsic task sender storage: {:?}", e)
+	})?;
+	let mut calls = vec::Vec::new();
+
+	loop {
+		let start_time = time::Instant::now();
+		while start_time.elapsed() < BATCH_EXTRINSIC_INTERVAL && calls.len() < MAX_BATCH_SIZE {
```

Comment:
Based on this condition `while start_time.elapsed() < BATCH_EXTRINSIC_INTERVAL && calls.len() < MAX_BATCH_SIZE {`, In high traffic, We are still going submit extrinsics in batches of 500 to the chain and take up most of the available block weight. 
- We need to decouple these two conditions, When we hit max_batch_size we need to maintain an overflow vector and push it there 
- We should submit extrinsic only when we hit the interval, and the max_batch_size should be such that it takes up less than half of the total block weight so that during high traffic the chain is still usable for other activities 
- These extrinsics are not criticial to the system or UX, So it's fine if we submit only one batch every interval, the case in point is that the chain should be usable and there should be a decent amount of block weight left after it. 
- Calculating the weight will be a difficult thing, But since this is a batch of the same calls, it might be easier to come up with an okay number, the goal is to not spam the chain with so many calls, so it doesn't have to be very accurate if it does the job.

Reply:
Oh yes, the condition of the while loop should only consider the interval, good catch. I think I had it that way, not sure why I changed it.

Reply:
if we take as an example (assuming the benchmarks are more or less accurate) the reference we have for `vc_issued` that is around 27M (the ref time), then 500 calls would take around 4% of the maxExtrinsic limit for normal class dispatch on the litentry parachain (If I read this correctly)

```
{
  baseBlock: {
    refTime: 392,184,000
    proofSize: 0
  }
  maxBlock: {
    refTime: 500,000,000,000
    proofSize: 5,242,880
  }
  perClass: {
    normal: {
      baseExtrinsic: {
        refTime: 113,638,000
        proofSize: 0
      }
      maxExtrinsic: {
        refTime: 324,886,362,000
        proofSize: 3,407,872
      }
      maxTotal: {
        refTime: 375,000,000,000
        proofSize: 3,932,160
      }
      reserved: {
        refTime: 0
        proofSize: 0
      }
    }
    operational: {
      baseExtrinsic: {
        refTime: 113,638,000
        proofSize: 0
      }
      maxExtrinsic: {
        refTime: 449,886,362,000
        proofSize: 4,718,592
      }
      maxTotal: {
        refTime: 500,000,000,000
        proofSize: 5,242,880
      }
      reserved: {
        refTime: 125,000,000,000
        proofSize: 1,310,720
      }
    }
    mandatory: {
      baseExtrinsic: {
        refTime: 113,638,000
        proofSize: 0
      }
      maxExtrinsic: null
      maxTotal: null
      reserved: null
    }
  }
}
```

I'm not 100% sure about these numbers but I think it should be fine. @Kailai-Wang Do you have some inputs on this? Is my assumption correct?

Reply:
500 should be a conservative estimation - we saw 1000+ `vc_issued` ext in one block without problems.

It's rather for (yet unknown) ext in the future if we plan to use this global thread to interact with parachain - but again it's not this PR's business